src/cmap/auth/mongodb_aws.ts

```diff
@@ -207,12 +217,29 @@ function deriveRegion(host: any) {
   return parts[1];
 }
 
-function request(uri: any, options: any, callback?: Function) {
-  if (typeof options === 'function') {
-    callback = options;
+interface RequestOptions {
+  json?: boolean;
+  method?: string;
+  timeout?: number;
+  headers?: {
```

Comment:
nit: given this is a pretty generic HTTP client, headers would likely be `{ [key: string]: any }`